### PR TITLE
Lockdown es index usage

### PR DESCRIPTION
**Pull request by wleese**
An addition to https://github.com/grafana/grafana/issues/5880

This commit does the following:

- allow only POSTs on _msearch and GETs to .*/_stats (assumption that Grafana doesn't need anything else, based on testing adding a data source, annotations and a graph metric)
- disallow any index specified in the POST body that does not correspond with the configured index pattern in the datasource

### CODE REVIEW

**Comment by torkelo** [comment]
a lot of code specific to ES in the dataproxy file now. We might just move all ES code to the tsdb backend layer (in alerting-elasticsearch branch, https://github.com/grafana/grafana/tree/alerting-elasticsearch) have have the frontend reuse that ES query path.

**Comment by wleese** [comment]
Is there any point in this being ported to the tsdb backend layer in alerting-elasticsearch branch right now, or shall I wait until the dust has settled?

**Comment by torkelo** [comment]
it might be a bit early. We are some months of having the time to refactor so the Data sources that support alerting reuse that backend code in the frontend data source component. But that is the goal.

File: pkg/api/dataproxy.go [hunk 1]
```diff
@@ -116,17 +119,43 @@ func ProxyDataSourceRequest(c *middleware.Context) {
 	}
 
 	if ds.Type == m.DS_ES {
-		if c.Req.Request.Method == "DELETE" {
```
Comment (torkelo) [inline pkg/api/dataproxy.go]: this code block that handles the ES data proxy needs to be broken out into separate component with tests, having all this ES code in side this function feels like a bad idea.

**Comment by bergquist** [comment]
Closing this due to inactivity.

ping me if you want to take a new stab at this and I will reopen.

**Pull request closed without merging**